.github/workflows/main.yml

```diff
@@ -35,28 +35,46 @@ jobs:
           - "3.4"
           - ruby-head
           - jruby-9.4
-        task:
-          - internal_investigation
-          - spec
-    name: "Ruby ${{ matrix.ruby }}: ${{ matrix.task }}"
+        parser_engine:
+          - parser_whitequark
+        include:
+          - ruby: "2.7" # Minimum version required for Prism to run.
+            parser_engine: parser_prism
+    name: "Ruby ${{ matrix.ruby }} - Spec (${{ matrix.parser_engine }})"
     steps:
       - uses: actions/checkout@v4
       - uses: ruby/setup-ruby@v1
         with:
           ruby-version: "${{ matrix.ruby }}"
           bundler-cache: true
-      - run: NO_COVERAGE=true bundle exec rake ${{ matrix.task }}
+      - run: bundle exec rake spec
+        env:
+          PARSER_ENGINE: ${{ matrix.parser_engine }}
+      - name: Upload Coverage Artifact
```

Comment:
I think we're quite fine if it just fails.
Also, it's fine if it runs for just one Ruby version, which is done below in the "coverage" job (line 50)

@rubocop/rubocop-rspec WDYT?

Reply:
> I think we're quite fine if it just fails.

Yeah, I can do that. `if-no-files-found` was a bit nice during debuging but not really necessary anymore (plus jruby kinda needs that, or something similar)